presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java

```diff
@@ -35,12 +35,15 @@
 import io.prestosql.spi.connector.ConnectorTableMetadata;
```

Comment:
What is the motivation behind this? What is the difference if engine would apply predicate or if predicate is applied by connector?

Reply:
It may be beneficial for testing the optimizer on plans where the whole TupleDomain gets pushed-down into the connector, as well as being a simple example for how to implement predicate pushdown.
Not sure if it worth it though...

Reply:
I can close this PR, if this doesn't worth the additional complexity.
Please let me know your opinion.

Reply:
> It may be beneficial for testing

Can you validate that and check if it actually beneficial or not?

I see memory connector already implements limit push down, so maybe I am missing something.

Reply:
I have used it for manually testing regular predicate pushdown together with dynamic filtering - I can add the test to `TestMemorySmoke` in this PR.
Also, I've used it for the debugging and testing of https://github.com/prestosql/presto/pull/844 (adding support for estimating statistics of pushed-down predicates).